vyper/semantics/types/user.py

```diff
@@ -281,9 +282,59 @@ def from_EventDef(cls, base_node: vy_ast.EventDef) -> "EventT":
         return cls(base_node.name, members, indexed, base_node)
 
     def _ctor_call_return(self, node: vy_ast.Call) -> None:
-        validate_call_args(node, len(self.arguments))
-        for arg, expected in zip(node.args, self.arguments.values()):
-            validate_expected_type(arg, expected)
+        # Handle positional args by converting them to kwargs
+        # TODO: Remove block when positional args are removed
+        if len(node.args) > 0:
+            if len(node.keywords) > 0:
+                # can't mix args and kwargs
+                raise InstantiationException(
+                    "Event instantiation requires either all positional arguments "
+                    "or all keyword arguments",
+                    node,
+                )
+
+            msg = "Instantiating events with positional arguments is "
+            msg += "deprecated as of v0.4.1 and will be disallowed "
+            msg += "in a future release. Use kwargs instead eg. "
+            msg += "Foo(a=1, b=2)"
+
+            vyper_warn(msg, node)
+
+            # convert positional args to keywords
+            kw_list = []
+            for kw, val in zip(self.arguments.keys(), node.args):
+                kw_node = vy_ast.keyword(arg=kw, value=val)
+                kw_node.value.set_parent(kw_node)  # type: ignore
```

Comment:
I think this line is unnecessary.

Reply:
you're right! only the other `set_parent` call is needed (or else source maps break), I removed this extraneous line